src/lib/integration_test_local_engine/docker_node_config.ml

```diff
@@ -0,0 +1,570 @@
+open Core_kernel
+open Async
+open Integration_test_lib
+open Docker_compose
+
+module PortManager = struct
+  let mina_internal_rest_port = 3085
+
+  let mina_internal_client_port = 8301
+
+  let mina_internal_metrics_port = 10001
+
+  let mina_internal_server_port = 3086
+
+  let mina_internal_external_port = 10101
+
+  let postgres_internal_port = 5432
+
+  type t =
+    { mutable available_ports : int list
+    ; mutable used_ports : int list
+    ; min_port : int
+    ; max_port : int
+    }
+
+  let create ~min_port ~max_port =
+    let available_ports = List.range min_port max_port in
+    { available_ports; used_ports = []; min_port; max_port }
+
+  let allocate_port t =
+    match t.available_ports with
+    | [] ->
+        failwith "No available ports"
+    | port :: rest ->
+        t.available_ports <- rest ;
+        t.used_ports <- port :: t.used_ports ;
+        port
+
+  let allocate_ports_for_node t =
+    let rest_port_source = allocate_port t in
+    let client_port_source = allocate_port t in
+    let metrics_port_source = allocate_port t in
+    [ { Dockerfile.Service.Port.published = rest_port_source
+      ; target = mina_internal_rest_port
+      }
+    ; { published = client_port_source; target = mina_internal_client_port }
+    ; { published = metrics_port_source; target = mina_internal_metrics_port }
+    ]
+
+  let release_port t port =
+    t.used_ports <- List.filter t.used_ports ~f:(fun p -> p <> port) ;
+    t.available_ports <- port :: t.available_ports
+
+  let get_latest_used_port t =
+    match t.used_ports with [] -> failwith "No used ports" | port :: _ -> port
+end
+
+module Base_node_config = struct
+  type t =
+    { peer : string option
+    ; log_level : string
+    ; log_snark_work_gossip : bool
+    ; log_txn_pool_gossip : bool
+    ; generate_genesis_proof : bool
+    ; client_port : string
+    ; rest_port : string
+    ; external_port : string
+    ; metrics_port : string
+    ; runtime_config_path : string option
+    ; libp2p_key_path : string
+    ; libp2p_secret : string
+    }
+  [@@deriving to_yojson]
+
+  let container_runtime_config_path = "/root/runtime_config.json"
+
+  let container_entrypoint_path = "/root/entrypoint.sh"
+
+  let container_keys_path = "/root/keys"
+
+  let container_libp2p_key_path = container_keys_path ^ "/libp2p_key"
+
+  let entrypoint_script =
+    ( "entrypoint.sh"
+    , {|#!/bin/bash
+  # This file is auto-generated by the local integration test framework.
+  # Path to the libp2p_key file
+  LIBP2P_KEY_PATH="|}
+      ^ container_libp2p_key_path
+      ^ {|"
+  # Generate keypair and set permissions if libp2p_key does not exist
```

Comment:
I define custom entry points for the daemon, postgres, and archive node containers. Would these be better served in a directory and removed from the inline definition? Where is the best location for these to live if we wanted to move them?